app/lib/actions/katello/host/upload_profiles.rb

```diff
@@ -38,7 +38,7 @@ def self.upload_modules_to_pulp(available_streams, host)
             ::Katello::ModuleStream.where(profile.slice(:name, :stream))
           end
 
-          query_name_streams = query_name_streams.inject(&:or)
+          query_name_streams = query_name_streams.inject(&:or) unless query_name_streams.empty?
```

Comment:
Something like this might be better ?
```diff
diff --git a/app/lib/actions/katello/host/upload_profiles.rb b/app/lib/actions/katello/host/upload_profiles.rb
index a17b68017a..6405a6b881 100644
--- a/app/lib/actions/katello/host/upload_profiles.rb
+++ b/app/lib/actions/katello/host/upload_profiles.rb
@@ -37,16 +37,18 @@ module Actions
           query_name_streams = available_streams.map do |profile|
             ::Katello::ModuleStream.where(profile.slice(:name, :stream))
           end
+          updated_profiles = []
+          if query_name_streams
+            query_name_streams = query_name_streams.inject(&:or)
 
-          query_name_streams = query_name_streams.inject(&:or)
+            bound_library_instances = host.content_facet.bound_repositories.map(&:library_instance_or_self)
+            query = ::Katello::ModuleStream.in_repositories(bound_library_instances).
+                                            select(:name, :stream, :version, :context, :arch).
+                                              merge(query_name_streams)
 
-          bound_library_instances = host.content_facet.bound_repositories.map(&:library_instance_or_self)
-          query = ::Katello::ModuleStream.in_repositories(bound_library_instances).
-                                          select(:name, :stream, :version, :context, :arch).
-                                            merge(query_name_streams)
-
-          updated_profiles = query.map do |module_stream|
-            module_stream.slice(:name, :stream, :version, :context, :arch)
+            updated_profiles = query.map do |module_stream|
+              module_stream.slice(:name, :stream, :version, :context, :arch)
+            end
           end
 
           # We also need to pass module streams that are not found in the ModuleStream table
```

Reply:
@parthaa  I think we'd still have to do `unless query_name_streams.empty?` since empty array is truthy